### PR TITLE
feat(node): Show IP address in node summary network card (#2851)

### LINKED ISSUE #2851: [Build] Add IP address to the networking card
from #2601

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Added interface IP address to node summary network card
- Moved MAC address beneath Name
- Added subnet beneath IP address
- Added vlan beneath Fabric

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to the summary page for a device
- Ensure the network card shows IP addresses for interfaces in the right column (may show "DHCP" or "Unconfigured")
- Ensure a subnet is shown beneath an IP address (if one is set for that interface)
- Ensure the MAC address is shown beneath interface names
- Ensure the VLAN (if set) is shown beneath fabrics

## Fixes

Fixes #2851 

## Launchpad issue

[lp#1881560](https://bugs.launchpad.net/maas/+bug/1881560)

Only bug and security fixes should be backported, new features should only land in main.

### Before 
image

### After
image

### CODE REVIEW

**Comment by sparkiegeek** [comment]
that After example looks odd - "DHCP", "10 Gbps", ... "No DHCP" ?

**Comment by ndv99** [comment]
> that After example looks odd - "DHCP", "10 Gbps", ... "No DHCP" ?

I was thinking the same thing, was actually hoping for some feedback on this. The function used in the IP address column shows an IP if one is set, "DHCP" if it gets its IPs through DHCP, and "Unconfigured" otherwise (as far as I know). Correct me if I'm wrong, but is the DHCP column saying whether or not that interface is a *provider* of DHCP? If so we could probably change the phrasing there for clarity

File: src/app/base/components/NodeSummaryNetworkCard/NetworkCardTable/NetworkCardTable.tsx [hunk 1]
```diff
@@ -47,19 +71,45 @@ const NetworkCardInterface = ({ interfaces }: Props): JSX.Element => {
             : null;
           const dhcpStatus = getDHCPStatus(vlan, vlans, fabrics);
 
+          console.log(iface.links);
```
Comment (sparkiegeek) [inline src/app/base/components/NodeSummaryNetworkCard/NetworkCardTable/NetworkCardTable.tsx:74]:
```suggestion
```

Comment (sparkiegeek) [reply src/app/base/components/NodeSummaryNetworkCard/NetworkCardTable/NetworkCardTable.tsx:74]: leftover debug?

Comment (ndv99) [reply src/app/base/components/NodeSummaryNetworkCard/NetworkCardTable/NetworkCardTable.tsx:74]: Good spot, thanks!

**Comment by petermakowski** [comment]
I think I found an issue. `getInterfaceSubnet` returns an object which cannot be rendered by React. Could you please look into this.

To reproduce: Go to machine details, click on the Network link in the Network card, click the native "go back" button in the browser.

```
Uncaught Error: Objects are not valid as a React child (found: object with keys {id, created, updated, name, description, vlan, cidr, rdns_mode, gateway_ip, dns_servers, allow_dns, allow_proxy, active_discovery, managed, disabled_boot_architectures, statistics, version, space}). If you meant to render a collection of children, use an array instead.
```

**Comment by ndv99** [comment]
> I think I found an issue. `getInterfaceSubnet` returns an object which cannot be rendered by React. Could you please look into this.
> 
> To reproduce: Go to machine details, click on the Network link in the Network card, click the native "go back" button in the browser.
> 
> ```
> Uncaught Error: Objects are not valid as a React child (found: object with keys {id, created, updated, name, description, vlan, cidr, rdns_mode, gateway_ip, dns_servers, allow_dns, allow_proxy, active_discovery, managed, disabled_boot_architectures, statistics, version, space}). If you meant to render a collection of children, use an array instead.
> ```

Sure, can do. Thanks for spotting that!

File: src/app/base/components/NodeSummaryNetworkCard/NodeSummaryNetworkCard.test.tsx [hunk 2]
```diff
@@ -73,27 +67,25 @@ describe("NodeSummaryNetworkCard", () => {
...
       "Product 1"
     );
-    expect(
-      wrapper.find("ul [data-testid='nic-firmware-version']").at(0).text()
-    ).toBe("1.0.0");
-    expect(wrapper.find("ul [data-testid='nic-vendor']").at(1).text()).toBe(
+    expect(screen.getAllByTestId("nic-firmware-version")[0]).toHaveTextContent(
+      "1.0.0"
+    );
+    expect(screen.getAllByTestId("nic-vendor")[1]).toHaveTextContent(
```
Comment (petermakowski) [inline src/app/base/components/NodeSummaryNetworkCard/NodeSummaryNetworkCard.test.tsx:88]: 👍 for migrating to RTL as you go

**Review by petermakowski — APPROVED** [review]

**Pull request merged**